cronjobs/ftp_connection.py

```diff
@@ -0,0 +1,88 @@
+# Python 3 module
+# -*- coding: utf-8 -*-
+
+from ftplib import FTP
+import os
+
+
+class FTPConnection:
+
+    _host = ""
+    _username = ""
+    _password = ""
+    _ftp = None
+
+    def __init__(self, host, username, password):
+        self._host = host
+        self._username = username
+        self._password = password
+        self._connect()
+        self._login()
+
+    def getFTP():
+        return _ftp
+
+    def _connect(self):
+        try:
+            self._ftp = FTP(host=self._host, timeout=120)
+        except Exception as e:
+            Error("failed to connect to FTP server! (" + str(e) + ")")
+
+    def _login(self):
+        try:
+            self._ftp.login(user=self._username, passwd=self._password)
+        except Exception as e:
+            Error("failed to login to FTP server! (" + str(e) + ")")
+
+    def changeDirectory(self, remote_dir_path):
+        error_message = "failed to change directory: " + remote_dir_path
+        try:
+            res = self._ftp.cwd(remote_dir_path)
+            if not res.startswith('250 '):
+                Error(error_message)
+        except Exception as e:
+            Error(error_message + " (" + str(e) + ")")
+
+    def listDirectory(self):
+        error_message = "failed to list directory"
+        try:
+            return self._ftp.nlst()
+        except Exception as e:
+            Error(error_message + " (" + str(e) + ")")
+
+    def downloadFile(self, remote_file_name, local_file_path):
+        try:
+            output = open(local_file_path, "wb")
+        except Exception as e:
+            Error("local open of \"" + local_file_path + "\" failed! (" + str(e) + ")")
+        try:
+            def RetrbinaryCallback(chunk):
+                try:
+                    output.write(chunk)
+                except Exception as e:
+                    Error("failed to write a data chunk to local file \"" + local_file_path + "\"! (" + str(e) + ")")
+            self._ftp.retrbinary("RETR " + remote_file_name, RetrbinaryCallback)
+        except Exception as e:
+            Error("File download failed! (" + str(e) + ")")
+
+
+    def uploadFile(self, local_file_path, remote_file_name=None):
+        if remote_file_name is None:
+            remote_file_name = os.path.basename(local_file_path)
+        error_message = "failed to upload file " + local_file_path + " to " + remote_file_name
+        try:
+            with open(local_file_path, 'rb') as fp:
+                res = self._ftp.storbinary("STOR " + remote_file_name, fp)
+                if not res.startswith('226 '):
+                    Error(error_message)
+        except Exception as e:
+            Error(error_message + " (" + str(e) + ")")
+
+    def renameFile(self, remote_file_name_old, remote_file_name_new):
```

Comment:
Maybe call this renameRemoteFile?

Reply:
Why? The only file you can rename in FTP context is the remote file. Providing a method inside a FTP class to rename a local file doesn't make sense in my opinion.

Reply:
Does everybody know this?

Reply:
Yes 😆

Reply:
I see...